File: core/src/com/lastwords/ashley/texture/TextureComponent.java
package com.lastwords.ashley.texture;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

public class TextureComponent implements Component {

    public TextureRegion textureRegion;

    public TextureComponent() {}

    public TextureComponent(TextureRegion textureRegion) {
        this.textureRegion = textureRegion;
    }

}

File: core/src/com/lastwords/ashley/texture/TextureComponent.kt
package com.lastwords.ashley.texture

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.graphics.g2d.TextureRegion

class TextureComponent(
        var textureRegion: TextureRegion? = null,
        var xOffset: Float = 0f,
        var yOffset: Float = 0f
) : Component
